src/Traits/HasConfiguration.php

```diff
@@ -97,27 +97,55 @@ public function withAuthTokenKeyName(string $authTokenKeyName): static
         return $this;
     }
 
+    public function withAutoIncrementingState(): static
```

Comment:
tbh, not sure if it's a good idea to change the naming of public functions in a minor release.
It seems to make sense, but I would keep that for a major release if there's no urgency behind that.

Reply:
What do you mean with changing the naming? I'm only adding functions.

Reply:
Sorry, my bad. I thought you replaced the `without` methods with `with`. All good 👌